### PR TITLE
chore(webhooks): update webhook function refactor

### PR DESCRIPTION
**Pull request by alemagio**
## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/server/modules/webhooks/domain/operations.ts [hunk 1]
```diff
@@ -3,12 +3,26 @@ import { Webhook } from '@/modules/webhooks/domain/types'
...
 }: Pick<Webhook, 'streamId'>) => Promise<number>
 
 export type GetWebhookById = ({ id }: Pick<Webhook, 'id'>) => Promise<Webhook | null>
+
+export type UpdateWebhook = ({
+  webhookId,
+  webhookInput
+}: {
+  webhookId: string
+  webhookInput: Pick<Webhook, 'updatedAt'> &
```
Comment (gjedlicska) [inline packages/server/modules/webhooks/domain/operations.ts:21]: updatedAt, is prob not needed here, it should be the service's responsibility to craft that date properly imho

**Review by gjedlicska — CHANGES_REQUESTED** [review]

File: packages/server/assets/webhooks/typedefs/webhooks.graphql [hunk 2]
```diff
@@ -84,7 +84,7 @@ input WebhookUpdateInput {
   description: String
   secret: String
   enabled: Boolean
-  triggers: [String]
+  triggers: [String!]
```
[the remark this answers is not in the document]
Comment (alemagio) [reply packages/server/assets/webhooks/typedefs/webhooks.graphql:87]: By doing that we want always an array, we don't accept undefined or null
which I don't think it's the case for the current FE implementation right?

I mean, we accept:
- null
- undefined
- array with non null or undefined value

https://stackoverflow.com/a/46771890

File: packages/server/modules/webhooks/domain/operations.ts (hunk 1 above)
Comment (alemagio) [reply packages/server/modules/webhooks/domain/operations.ts:21]: yes, that's why it is required.
I mean the type says that the `updatedAt` field is required.
Does it make sense?

File: packages/server/assets/webhooks/typedefs/webhooks.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (gjedlicska) [reply packages/server/assets/webhooks/typedefs/webhooks.graphql:87]: disregard this, now i now, this is the update, we don't need to for this to be defined

File: packages/server/modules/webhooks/repositories/webhooks.ts [hunk 3]
```diff
@@ -55,3 +56,21 @@ export const getWebhookByIdFactory =
 
     return { ...webhook, triggers: toTriggersArray(webhook.triggers) }
   }
+
+export const updateWebhookFactory =
+  ({ db }: { db: Knex }): UpdateWebhook =>
+  async ({ webhookId, webhookInput }) => {
+    const { triggers, ...update } = webhookInput
+    if (triggers) {
+      ;(
```
Comment (gjedlicska) [inline packages/server/modules/webhooks/repositories/webhooks.ts:65]: this looks very confusing

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**